classes/question/bank/question_bank_filter.php

```diff
@@ -444,6 +444,27 @@ public function get_sql_filter($data) {
         }
         return array($res, $params);
     }
+
+    public function check_data($formdata) {
+        $data = parent::check_data($formdata);
+
+        if ((!is_numeric($data['value']) && !empty($data['value'])) ||
+                $data['value'] < 0 || $data['value'] > PHP_INT_MAX) {
+            return false;
+        }
+
+        return $data;
+    }
+
+    public function setupForm(&$mform) {
```

Comment:
It has not been necessary to pass objects be reference in PHP since PHP 5.0. Can we delete the '&' please. (Applies in several places throughout the patch.)

Reply:
Also, this is causing CodeChecker to fail. You have to use an invalid method name because of the parent class, but please tell CodeChecker that: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-parts-of-a-file